### PR TITLE
feat: introduce related articles component for next theme

### PR DESCRIPTION
**Pull request by dcshzj**
Figma: https://www.figma.com/file/8TEzSSDbx97gctRJ7nKdqG/Website-Template-Gen-2?type=design&node-id=1903-8989&mode=design&t=Xc04DSeQByKZOIJg-4

Not yet done (will do in separate PR): Determine what exactly is a related article

### CODE REVIEW

File: src/templates/next/components/shared/Siderail/Siderail.tsx [hunk 1]
```diff
@@ -50,13 +51,13 @@ const SiderailMobile = ({
                 )}
                 {isCurrent && (
                   <ul>
-                    {childPages?.map(({ title, url }, index) => {
+                    {childPages?.map(({ title, url }, childIndex) => {
                       return (
-                        <li>
+                        <li key={childIndex}>
```
Comment (harishv7) [inline src/templates/next/components/shared/Siderail/Siderail.tsx:56]: usually we don't use indexes as keys right?

File: src/templates/next/layouts/Article/Article.tsx [hunk 2]
```diff
@@ -37,15 +38,33 @@ const ArticleLayout = ({
...
           <div className="overflow-x-auto w-full lg:max-w-[660px]">
             {content.map((component) =>
               renderComponent({ component, LinkComponent }),
             )}
           </div>
 
           <div className="w-full lg:max-w-[260px]">
-            <p>Placeholder for related articles component</p>
+            <RelatedArticles
+              items={[
```
Comment (harishv7) [inline src/templates/next/layouts/Article/Article.tsx:50]: why is this hardcoded?

File: src/templates/next/layouts/Article/Article.tsx [hunk 3]
```diff
@@ -37,15 +38,33 @@ const ArticleLayout = ({
...
+                  title:
+                    "Wild boar escapes Singapore Zoo; captivated and returned to safety within 3 hours",
+                  url: "/items/second",
+                },
+                {
+                  title: "Mynah population decreasing YoY, a worrying trend",
+                  url: "/items/third",
+                },
+              ]}
+              LinkComponent={LinkComponent}
```
Comment (harishv7) [inline src/templates/next/layouts/Article/Article.tsx:66]: is the default for LinkComponent set to be `a` tag?

**Comment by dcshzj** [comment]
Closing this for now, we opted not to have this component yet. Will do a separate PR that updates the article layout.

**Pull request closed without merging**